Null node can cause segfault

Document:
```yaml
plip:
  plop:
    plaf: ~
```

Patch:
```yaml
- path: /plip/plop/plaf/plaf
  op: remove
```

Result:
```
$ ./yaml-patch -o test-patch2.yml < test-doc2.yml 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x113184c]

goroutine 1 [running]:
github.com/krishicks/yaml-patch.tryRemove(0x1220300, 0xc42008a088, 0xc4200864c0, 0xffffffffffffffff, 0xc4200d7c58)
	/home/pilot/go/src/github.com/krishicks/yaml-patch/operation.go:101 +0x10c
github.com/krishicks/yaml-patch.(*Operation).Perform(0xc4200864c0, 0x1220300, 0xc42008a088, 0x11466c0, 0xc42009a760)
	/home/pilot/go/src/github.com/krishicks/yaml-patch/operation.go:70 +0x318
github.com/krishicks/yaml-patch.Patch.Apply(0xc420086400, 0x1, 0x1, 0xc4200f0600, 0x52, 0x600, 0x600, 0x1, 0x17, 0xc4200984b0, ...)
	/home/pilot/go/src/github.com/krishicks/yaml-patch/patch.go:52 +0x56c
main.main()
	/home/pilot/go/src/github.com/krishicks/yaml-patch/cmd/yaml-patch/main.go:50 +0x58c
```

Tested with release `v0.0.10` on Darwin.

I'll really skeptical about [these lines of code](https://github.com/krishicks/yaml-patch/blob/master/pathfinder.go#L67-L71): how could a path match a container or its parent?

--- a/pathfinder.go
+++ b/pathfinder.go
@@ -62,13 +62,9 @@ func find(part string, routes map[string]Container) map[string]Container {
 			continue
 		}
 
-		if node, err := container.Get(part); err == nil {
+		if node, err := container.Get(part); err == nil && node != nil {
 			path := fmt.Sprintf("%s/%s", prefix, part)
-			if node == nil {
-				matches[path] = container
-			} else {
-				matches[path] = node.Container()
-			}
+			matches[path] = node.Container()
 		}
 	}
 

--- a/pathfinder_test.go
+++ b/pathfinder_test.go
@@ -59,7 +59,6 @@ jobs:
 			Entry("return a route for a single submatch with help", "/jobs/get=A/args/arg=arg2", []string{"/jobs/0/plan/0/args/1"}),
 			Entry("return a route for a single submatch with no help", "/jobs/get=A/arg=arg2", []string{"/jobs/0/plan/0/args/1"}),
 			Entry("return a route for a single submatch with help using escape ordering", "/jobs/get=C~1D", []string{"/jobs/0/plan/2"}),
-			Entry("return a route when given a pointer with a leaf that does not exist", "/jobs/name=job1/nonexistent", []string{"/jobs/0/nonexistent"}),
 			Entry("return a route when given a pointer with an array thingy", "/jobs/name=job1/plan/-", []string{"/jobs/0/plan/-"}),
 		)
 		DescribeTable(
@@ -69,6 +68,8 @@ jobs:
 			},
 			Entry("return any routes when given a bad index", "/jobs/2"),
 			Entry("return any routes when given a bad index", "/jobs/-1"),
+			Entry("return any routes when given a badly nested index", "/jobs/0/foo/bar/baz"),
+			Entry("return a route when given a pointer with a leaf that does not exist", "/jobs/name=job1/nonexistent"),
 		)
 	})
 })

